frontend/src/app/plans/page.tsx

```diff
@@ -1,49 +1,32 @@
-"use client";
+import React from "react";
 
-import { atom, useAtom } from "jotai";
-import { PlusIcon } from "lucide-react";
-import { useRouter } from "next/navigation";
+import type { ExtendedCourse } from "@/atoms/planFamily";
+import { fetchToAdonis } from "@/lib/auth";
+import type { Registration } from "@/lib/types";
 
-import { planFamily } from "@/atoms/planFamily";
-import { plansIds } from "@/atoms/plansIds";
-import { PlanItem } from "@/components/PlanItem";
+import { PlansPage } from "./_components/PlansPage";
 
-const plansAtom = atom(
-  (get) => get(plansIds).map((id) => get(planFamily(id))),
-  (get, set, values: Array<{ id: string }>) => {
-    set(plansIds, values);
-  },
-);
-export default function Plans() {
-  const [plans, setPlans] = useAtom(plansAtom);
-  const router = useRouter();
-  const addNewPlan = () => {
-    const uuid = crypto.randomUUID();
-    const newPlan = {
-      id: uuid,
-    };
+export interface PlanResponseDataType {
+  id: number;
+  userId: number;
+  name: string;
+  createdAt: string;
+  updatedAt: string;
+  courses: ExtendedCourse[];
+  registrations: Registration[];
+}
+
+export interface ErrorResponse {
+  error: string;
+}
 
-    void window.umami?.track("Create plan", {
-      numberOfPlans: plans.length,
-    });
+type PlanResponse = ErrorResponse | PlanResponseDataType[];
 
-    router.push(`/plans/create/${newPlan.id}`);
-    setPlans([...plans, newPlan]);
-  };
+export default async function Plans() {
+  const data = await fetchToAdonis<PlanResponse>({
+    url: "/user/schedules",
+    method: "GET",
+  });
 
-  return (
-    <div className="container mx-auto max-h-full flex-1 flex-grow overflow-y-auto p-4">
-      <div className="grid grid-cols-2 gap-4 sm:justify-start md:grid-cols-3 lg:grid-cols-5 xl:grid-cols-6">
-        <button
-          onClick={addNewPlan}
-          className="group flex aspect-square items-center justify-center rounded-lg border-2 border-dashed border-gray-400 p-4 shadow-md transition-all hover:border-primary hover:bg-primary/5 hover:shadow-xl"
-        >
-          <PlusIcon className="h-24 w-24 text-gray-400 transition-colors group-hover:text-primary" />
-        </button>
-        {plans.map((plan) => (
-          <PlanItem key={plan.id} id={plan.id} name={plan.name} />
-        ))}
-      </div>
-    </div>
-  );
+  return <PlansPage plans={Array.isArray(data) ? data : []} />;
```

Comment:
dziwne, że data nie może być arrayem, plus brak error handlingu tutaj

Reply:
dodaj error handling

Reply:
no ale po co, w sensie jak error czyli jesli nie jestes zalogowany to ma zwrocic pusty array, i tyle

Reply:
jest trochę wiecej rodzajów błędów niż to jak nie jesteś zalogowany 😭